### PR TITLE
Implement isValidClaim

### LINKED ISSUE #22: Implement isValidClaim
`isValidClaim` should take a claim and validate the id, publicKey, and signature of the claim.

### PR DESCRIPTION
**Pull request by jwicks31**
Added isValidClaim: should take a claim and validate the id, publicKey, and signature of the claim.
Adjust testing format to try and answer the 5 Questions Every Unit Test Must Answer.

Fixes #22

### CODE REVIEW

File: src/Claim.ts [hunk 1]
```diff
@@ -72,3 +72,9 @@ export function createClaim(privateKey: string, type: ClaimType, attributes: Cla
     signature,
   }
 }
+
+export function isValidClaim(claim = {}): boolean {
+  if (!isClaim(claim)) return false
```
Comment (lautarodragan) [inline src/Claim.ts]: Don't you prefer the functional approach? `return isClaim(claim) && isValidSignature(claim) && getClaimId(claim) === claim.id`?

File: test/Claim/IsValidClaim.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  @Test('isValidClaim() : Given a valid cliam : should return true')
+  @TestCase(TheRaven)
+  public correctClaim(claim: Claim) {
+    Expect(isValidClaim(claim)).toBe(true)
+  }
+
+  @Test('isValidClaim() : Given an object that is not a claim : should return false')
+  @TestCase()
+  @TestCase({ foo: 'bar' })
+  public notClaimObject(claim: Claim) {
```
Comment (kennylavender) [inline test/Claim/IsValidClaim.ts]: @jwicks31 I forgot to have us change this to `notClaimObject(claim?: any)`

File: test/Claim/IsValidClaim.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
+/* tslint:disable:no-relative-imports */
+import { Expect, Test, TestCase } from 'alsatian'
+
+import { isValidClaim } from 'Claim'
+import { Claim } from 'Interfaces'
+//
```
Comment (kennylavender) [inline test/Claim/IsValidClaim.ts]: empty comment line should be removed

File: test/Claim/IsValidClaim.ts [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+import { TheRaven } from '../Claims'
+
+export class IsValidClaim {
+  @Test('isValidClaim() : Given a valid cliam : should return true')
```
Comment (kennylavender) [inline test/Claim/IsValidClaim.ts]: claim is misspelled in `Given a valid cliam`

File: src/Claim.ts (hunk 1 above)
Comment (kennylavender) [reply src/Claim.ts]: Yup, that looks better, I forgot to have us go back and refactor that ;)

Comment (jwicks31) [reply src/Claim.ts]: I had to do it slightly different. IsClaim returns undefined for object.id(and more keys) if it is not on the object that we pass in. 
`if (!isClaim(claim)) return false`
`return isValidSignature(claim) && getClaimId(claim) === claim.id`

File: test/Claim/IsValidClaim.ts [hunk 5]
```diff
@@ -0,0 +1,39 @@
+/* tslint:disable:no-relative-imports */
```
Comment (warrenv) [inline test/Claim/IsValidClaim.ts]: Is this a unit test? If so, let's move it to `src/Claim.test.ts`.

Comment (lautarodragan) [reply test/Claim/IsValidClaim.ts]: We already had in place unit tests for poet-js with this structure. We're sticking with it in this PR. We can move all of them in a separate PR

File: src/Claim.ts (hunk 1 above)
Comment (lautarodragan) [reply src/Claim.ts]: Hmm I'm not sure I understand. `isClaim(claim)` can return `undefined`? If so, it's still a falsey value, so it'll evaluate to true/false

File: test/Claim/IsValidClaim.ts (hunk 5 above)
Comment (wzalazar) [reply test/Claim/IsValidClaim.ts]: I agree with @lautarodragan, we have to do it in a separate PR

Comment (jwicks31) [reply test/Claim/IsValidClaim.ts]: Create a file for it? There are some other tests stored in the same folder as I made this one.

File: src/Claim.ts (hunk 1 above)
Comment (jwicks31) [reply src/Claim.ts]: It was making the notClaimObject test return undefined whenever I tried it that way.

Comment (kennylavender) [reply src/Claim.ts]: @jwicks31 try a double _not_ operator to coerce it to a boolean value. `!!isClaim(clam)``

**Comment by kennylavender** [comment]
@jwicks31 I am happy with the code now, except for the need to repeatedly hardcode the function name in `@test()` so that we can answer the 5 questions. But I still haven't had a good idea to improve that yet. Maybe someone else does?

**Comment by lautarodragan** [comment]
@warrenv the node uses these poet-js functions, so even if they don't use poet-js, these unit tests will impact the node. I'm not sure we want these tests in particular replicated to the other repo.

**Comment by kennylavender** [comment]
@lautarodragan neither @jwicks31 or I have access to merge the PR, if you could update that when you get a chance

**Comment by lautarodragan** [comment]
Try now please @kennylavender

**Comment by lautarodragan** [comment]
Just for documentation purposes, there was an issue with semantic release and it published over and old version. We're not sure whether it was actually able to overwrite version 1.0.0 or that version hadn't been published at all to begin with, but now version 1.0.0 is actually newer than 1.2.0. 
Walter's fixed this and now semantic release will start publishing with the correct version numbers, so just ignore the automated comment by semantic release (using Walter's account — we'll want to change that!) in this case.

**Pull request merged**